### PR TITLE
ci: add commitlint job

### PR DESCRIPTION
**Pull request by yshrsmz**
To enforce [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) described in [CONTRIBUTING.md](https://github.com/bucketeer-io/android-client-sdk/blob/main/CONTRIBUTING.md), I've added commitlint check to CI.

failed: https://github.com/yshrsmz/android-client-sdk/actions/runs/3094712074/jobs/5008378745
succeeded: https://github.com/yshrsmz/android-client-sdk/actions/runs/3094871512/jobs/5008697637

### CODE REVIEW

File: .github/workflows/pullrequest.yml [hunk 1]
```diff
@@ -45,5 +44,24 @@ jobs:
...
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+
+      - name: Install commitlint
+        run: npm install @commitlint/cli @commitlint/config-conventional
```
Comment (yshrsmz) [inline .github/workflows/pullrequest.yml:61]: https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional

File: .github/workflows/pullrequest.yml [hunk 2]
```diff
@@ -45,5 +44,24 @@ jobs:
           name: bucketeer-build-reports
           path: bucketeer/build/reports
 
+
+  commitlint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (yshrsmz) [inline .github/workflows/pullrequest.yml:53]: we need to clone the full repository to check commits in the PR.

**Review by cre8ivejp — APPROVED** [review]
Very good!
Thanks!

**Comment by yshrsmz** [comment]
thanks

**Pull request merged**